### PR TITLE
Refactor Bicep tool to use standalone CLI

### PR DESCRIPTION
**Pull request by wbreza**
Refactors the Bicep tool in `azd` to use the standalone API vs `az` command wrapper

### CODE REVIEW

File: cli/azd/pkg/tools/bicep/bicep.go [hunk 1]
```diff
@@ -93,33 +88,33 @@ func (cli *bicepCli) CheckInstalled(ctx context.Context) (bool, error) {
...
-			return "", fmt.Errorf("parsing cli version json: %s: %w", verRes.Stdout, err)
-		}
+	// 	if err := json.Unmarshal([]byte(verRes.Stdout), &jsonVer); err != nil {
+	// 		return "", fmt.Errorf("parsing cli version json: %s: %w", verRes.Stdout, err)
+	// 	}
 
-		return jsonVer.AzureCli, nil
-	}
+	// 	return jsonVer.AzureCli, nil
+	// }
```
Comment (wbreza) [inline cli/azd/pkg/tools/bicep/bicep.go:106]: Do we still need this check? If i recall the history of this issue it only errors out when called through `az` but the standalone worked correctly??  @ellismg ?

File: cli/azd/pkg/tools/bicep/bicep.go [hunk 2]
```diff
@@ -93,47 +89,22 @@ func (cli *bicepCli) CheckInstalled(ctx context.Context) (bool, error) {
...
-	args := []string{"bicep", "build", "--file", file, "--stdout"}
-
-	// Workaround azure/azure-cli#22621, by passing `--no-restore` to the CLI when
-	// when version 2.37.0 is installed.
-	if ver, err := sniffCliVersion(); err != nil {
-		log.Printf("error sniffing az cli version: %s", err.Error())
-	} else if ver == "2.37.0" {
-		log.Println("appending `--no-restore` to bicep arguments to work around azure/azure-dev#22621")
-		args = append(args, "--no-restore")
-	}
```
Comment (wbreza) [inline cli/azd/pkg/tools/bicep/bicep.go:122]: @ellismg Do we still need this check? If I recall the history of this issue the issue was with the `az cli` bicep wrapper and not bicep itself.

Comment (ellismg) [reply cli/azd/pkg/tools/bicep/bicep.go:122]: Oh, so sorry Walace - I was waiting for a CI job to run and ended up opening #1074 a few hours before you.  You are correct that this code is no longer needed, the bug was with the wrapper itself.

**Review by ellismg — APPROVED** [review]
LGTM

File: cli/azd/pkg/tools/bicep/bicep.go [hunk 3]
```diff
@@ -23,22 +23,16 @@ type BicepCli interface {
 
 func NewBicepCli(ctx context.Context) BicepCli {
 	return &bicepCli{
-		cli:           azcli.GetAzCli(ctx),
+		commandPath:   "bicep",
```
[the remark this answers is not in the document]
Comment (wbreza) [reply cli/azd/pkg/tools/bicep/bicep.go]: Updated to be `nil` by default and is set during first command invocation similar to @ellismg PR #1074 .

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**